backend/app/tests/anti_recommenders/open_ai/test_open_ai_normal_anti_recommender.py

```diff
@@ -0,0 +1,91 @@
+from langchain.schema.runnable import RunnableSequence, RunnableSerializable
+from pytest_mock import MockFixture
+
+from app.anti_recommenders.open_ai.open_ai_normal_anti_recommender import (
+    OpenAiNormalAntiRecommender,
+)
+from app.models.anti_recommendation import AntiRecommendation
+
+
+def test_build_chain(
```

Comment:
You typically don't test protected methods. If you want to test it locally for debugging purposes, you should add a pytest skip if `"CI" in os.environ`.

```
@pytest.mark.skipif("CI" in os.environ, reason="don't have OpenAI key in CI")
```

Reply:
I have added `skipif` decorators to tests for `NormalOpenAiAntiRecommender` and `AntiRecommendationEngine`. 
I will remove them from `AntiRecommendationEngine` once I add `AntiRecommender` implementations that can be run in the CI.